make/autoconf/toolchain.m4

```diff
@@ -620,6 +620,13 @@ AC_DEFUN_ONCE([TOOLCHAIN_DETECT_TOOLCHAIN_CORE],
     # All other toolchains use the compiler to link.
     LD="$CC"
     LDCXX="$CXX"
+    # Force use of lld, since that is what we expect when setting flags later on
+    if test "x$TOOLCHAIN_TYPE" = xclang; then
+      if test "x$OPENJDK_BUILD_OS" != "xmacosx"; then
```

Comment:
According to the implementation of makefile function `FLAGS_SETUP_LDFLAGS_HELPER` which locate in make/autoconf/flags-ldflags.m4 file, only the combanation of `$TOOLCHAIN_TYPE == clang` and `$OPENJDK_TARGET_OS == linux` support lld linker.

Reply:
@sendaoYan fyi, i've raised a new bug as this fails to build on AIX clang: https://bugs.openjdk.org/browse/JDK-8353658

Reply:
> @sendaoYan fyi, i've raised a new bug as this fails to build on AIX clang: https://bugs.openjdk.org/browse/JDK-8353658

Hi @andrew-m-leonard The new bug you mention seems duplicated to [JDK-8353005](https://bugs.openjdk.org/browse/JDK-8353005). Could you check that, thanks.